amethyst_animation/src/sprite.rs

```diff
@@ -0,0 +1,120 @@
+use amethyst_core::specs::prelude::Read;
+use amethyst_renderer::{SpriteRender, SpriteSheetSet};
+use minterpolate::InterpolationPrimitive;
+
+use {AnimationSampling, ApplyData, BlendMethod};
+
+/// Sampler primitive for SpriteRender animations
+/// Note that sprites can only ever be animated with `Step`, or a panic will occur.
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+pub enum SpriteRenderPrimitive {
+    SpriteSheet(u64),
+    SpriteIndex(usize),
+}
+
+impl InterpolationPrimitive for SpriteRenderPrimitive {
+    fn add(&self, _: &Self) -> Self {
+        panic!("Cannot add SpriteRenderPrimitive")
+    }
+
+    fn sub(&self, _: &Self) -> Self {
+        panic!("Cannot sub SpriteRenderPrimitive")
+    }
+
+    fn mul(&self, _: f32) -> Self {
+        panic!("Cannot mul SpriteRenderPrimitive")
+    }
+
+    fn dot(&self, _: &Self) -> f32 {
+        panic!("Cannot dot SpriteRenderPrimitive")
+    }
+
+    fn magnitude2(&self) -> f32 {
+        panic!("Cannot magnitude2 SpriteRenderPrimitive")
+    }
+
+    fn magnitude(&self) -> f32 {
+        panic!("Cannot magnitude SpriteRenderPrimitive")
+    }
+
+    fn normalize(&self) -> Self {
+        panic!("Cannot normalize SpriteRenderPrimitive")
+    }
```

Comment:
heh? Why implement it if you can't actually use the trait at all.

Reply:
because [it's a trait bound](https://github.com/amethyst/amethyst/blob/4cd5f4bdc218bd6278f71ba36b66a900c1c593e9/amethyst_animation/src/resources.rs#L29), [`MaterialPrimitive`](https://github.com/amethyst/amethyst/blob/4cd5f4bdc218bd6278f71ba36b66a900c1c593e9/amethyst_animation/src/material.rs#L15) does too; it's just that both of these animations aren't able to be interpolated, and must be stepped